yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -66,14 +66,16 @@ private[spark] class ApplicationMaster(
   // allocation is enabled), with a minimum of 3.
 
   private val maxNumExecutorFailures = {
-    val defaultKey =
+    val effectiveNumExecutors =
```

Comment:
Ok, let's just leave this alone. For users encountering the issue in Spark1.6, they can always manually set `spark.yarn.max.executor.failures` to a larger number as a workaround. Thanks @srowen . I'll close this.